### DISCUSSION TITLE
The stream or file "/var/www/html/storage/logs/ff3-cli-2022-02-10.log" could not be opened in append mode: Failed to open stream: Permission denied

### DISCUSSION THREAD

**Discussion by creylopez** [firefly-iii/firefly-iii#5759] [General]
Since some days ago I'm getting this error:

The stream or file "/var/www/html/storage/logs/ff3-cli-2022-02-10.log" could not be opened in append mode: Failed to open stream: Permission denied

(the file name corresponds to logs of the particular day in which the error is displayed)

Firefly says the error is unrecoverable.

The Ups! message is in Spanish but can be translated as "Ups! And error happened"

**Comment by creylopez** [comment]
By the way, it all started when I programmed a cron to be fired every night. I have checked the logs in storage/logs and it seems that has to be with permissions:
root@firefly-iii-cddd99df9-684t2:/var/www/html/storage/logs# ls -l
total 16
-rw-r--r-- 1 www-data www-data 3614 Feb  9 20:11 ff3-audit-2022-02-09.log
-rw-r--r-- 1 www-data www-data  486 Feb 10 06:58 ff3-audit-2022-02-10.log
-rw-r--r-- 1 www-data www-data 1130 Feb  9 20:11 ff3-cli-2022-02-09.log
-rw-r--r-- 1 root     root      836 Feb 10 00:05 ff3-cli-2022-02-10.log

if I change the affected file permissions to www-data then reloading the page works well.

I discovered this today. Before what I did was to restart container.

**Comment by JC5** [comment]
https://docs.firefly-iii.org/firefly-iii/faq/docker/#i-get-permission-denied-errors-on-the-cache-folder

**Reply by creylopez** [reply]
Thank you, that helped. Nevertheless, I'm using kubectl and the --user flag is not available.

What I've done is to use the runuser command:

$ kubectl exec -ti my-container -- runuser -u www-data /usr/local/bin/php /var/www/html/artisan firefly-iii:cron
